manticore/platforms/evm_world_state.py

```diff
@@ -0,0 +1,478 @@
+import logging
+from abc import ABC, abstractmethod
+from eth_typing import ChecksumAddress, URI
+from io import TextIOBase
+from typing import Dict, List, Optional, Set, Tuple, Union
+from urllib.parse import ParseResult, urlparse
+from web3 import Web3
+from ..core.smtlib import Array, ArrayVariable, BitVec, BitVecConstant, BitVecITE, BitVecZeroExtend, ConstraintSet
+from ..ethereum.state import State
+from ..exceptions import EthereumError
+
+logger = logging.getLogger(__name__)
+
+
+# sam.moelius: map records which (symbolic) offsets have been written to.  data holds the values
+# written.
+
+
+class Storage:
+    def __init__(self, constraints: ConstraintSet, address: int):
+        """
+        :param constraints: the ConstraintSet with which this Storage object is associated
+        :param address: the address that owns this storage
+        """
+        self.constraints = constraints
+        self.warned = False
+        self.data = constraints.new_array(
+            index_bits=256,
+            value_bits=256,
+            name=f"STORAGE_DATA_{address:x}",
+            avoid_collisions=True,
+            # sam.moelius: The use of default here creates unnecessary if-then-elses.  See
+            # ArrayProxy.get in expression.py.
+            # default=0,
+        )
+        self.dirty = False
```

Comment:
`self.dirty != ( len(self.data.written) !=0 )` ?

Reply:
I suspect they are are equal.

Accessing an object's data members feels wrong to me, but I take it you prefer that to having `dirty`?

Let me know.  I am happy to make the change.

Reply:
I am operating on the idea that if the member is not prepended with an underscore it is good for general use. :shrug: 
But other than that the question I'm trying to answer is why do we need Storage instead of ArrayProxy. (Other than its better naming)

Reply:
At this point, it looks likes Storage is just a thin wrapper around `ArrayProxy`. It adds a few methods and set some defaults in the constructor, but that's about it.